property/index.js

```diff
@@ -166,3 +156,68 @@ module.exports = yeoman.generators.Base.extend({
 
   saveProject: actions.saveProject
 });
+
+function coerceDefault(def, value) {
+  var isSupported = typeChoices.indexOf(def.type) !== -1;
+  debug('property type "%s" >> supported: %s', def.type, isSupported);
+
+  if (typeof value === 'string' && isSupported) {
+    switch ((def.type || '').toLowerCase()) {
+      case 'string':
+        if (value === 'uuid' || value === 'guid'){
+          def.defaultFn = value;
+        } else {
+          def.default = value;
+        }
+        break;
+      case 'number':
+        def.default = Number(value);
+        break;
+      case 'boolean':
+        if (['true', '1', 't'].indexOf(value) !== -1 ){
+          def.default = true;
+        } else {
+          def.default = false;
+        }
+        break;
+      case 'object':
+        def.default = JSON.parse(value);
+        break;
+      case 'array':
+        def.default = value.replace(/[\s,]+/g, ',').split(',');
```

Comment:
IIRC, when the user specifies `array` type, we ask them to provide `itemType` too. We should coerce array items of the default value depending on `itemType`. For example, when users selects "array of numbers" and provide default value `1,2`, then we should set the default value to `[1,2]` instead of `["1","2"]`.